compiler/src/org.graalvm.compiler.hotspot.jdk21.test/src/org/graalvm/compiler/hotspot/jdk21/test/SharedExceptionHandlerLoopTest.java

```diff
@@ -0,0 +1,191 @@
+     *             // same NPE handler as above. dispatch block outside loop
+     *             return null;
+     *         }
+     *     }
+     * }
+     * </pre>
+     */
+    @Override
+    protected byte[] generateClass(String internalClassName) {
+        ClassWriter cw = new ClassWriter(ClassWriter.COMPUTE_FRAMES);
+        cw.visit(Opcodes.V1_8, Opcodes.ACC_PUBLIC, internalClassName, null, "java/lang/Object", null);
+
+        MethodVisitor mv = cw.visitMethod(
+                        Opcodes.ACC_PUBLIC | Opcodes.ACC_STATIC,
+                        "testMethod",
+                        "(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;",
+                        null,
+                        new String[]{"java/lang/NullPointerException", "java/lang/IllegalAccessError", "java/lang/Exception"});
+        mv.visitCode();
+
+        Label start = new Label();
+        Label loopExcEnd = new Label();
+        Label loopNPEHandler = new Label();
+        Label loopIAEHandler = new Label();
+        Label loopEHandler = new Label();
+        Label afterLoopExcStart = new Label();
+        Label afterLoopExcEnd = new Label();
+        Label retLabel = new Label();
+
+        mv.visitTryCatchBlock(start, loopExcEnd, loopNPEHandler, "java/lang/NullPointerException");
+        mv.visitTryCatchBlock(start, loopExcEnd, loopIAEHandler, "java/lang/IllegalAccessError");
+        mv.visitTryCatchBlock(start, loopExcEnd, loopEHandler, "java/lang/Exception");
+        mv.visitTryCatchBlock(afterLoopExcStart, afterLoopExcEnd, loopNPEHandler, "java/lang/NullPointerException");
+
+        mv.visitLabel(start);
+        mv.visitVarInsn(Opcodes.ALOAD, 0);
+        mv.visitMethodInsn(
+                        Opcodes.INVOKEVIRTUAL,
+                        "java/lang/Object",
+                        "toString",
+                        "()Ljava/lang/String;",
+                        false);
+        mv.visitLabel(loopExcEnd);
+        mv.visitInsn(Opcodes.ARETURN);
+
+        mv.visitLabel(loopNPEHandler);
+        mv.visitVarInsn(Opcodes.ASTORE, 2);
+        mv.visitInsn(Opcodes.ACONST_NULL);
+        mv.visitInsn(Opcodes.ARETURN);
+
+        mv.visitLabel(loopIAEHandler);
+        mv.visitVarInsn(Opcodes.ASTORE, 2);
+        mv.visitJumpInsn(Opcodes.GOTO, start);
+
+        mv.visitLabel(loopEHandler);
+        mv.visitVarInsn(Opcodes.ASTORE, 2);
+        mv.visitJumpInsn(Opcodes.GOTO, retLabel);
+
+        mv.visitLabel(afterLoopExcStart);
+        mv.visitVarInsn(Opcodes.ALOAD, 1);
+        mv.visitMethodInsn(
+                        Opcodes.INVOKEVIRTUAL,
+                        "java/lang/Object",
+                        "toString",
+                        "()Ljava/lang/String;",
+                        false);
+        mv.visitLabel(afterLoopExcEnd);
+        mv.visitInsn(Opcodes.ARETURN);
+
+        mv.visitLabel(retLabel);
+        mv.visitInsn(Opcodes.ACONST_NULL);
+        mv.visitInsn(Opcodes.ARETURN);
```

Comment:
```suggestion
```

This looks like a duplicate of the `NPEHandler`, it doesn't exist in the original test. Is there a reason you introduce it here?